paddle/phi/kernels/fusion/cutlass/moe_kernel.cu

```diff
@@ -0,0 +1,920 @@
+    cudaStream_t stream) {
+  const int threads = 128;
+  const int blocks = num_experts;
+
+  initialize_expert_choice_route_kernel<<<blocks, threads, 0, stream>>>(
+      expert_for_source_row,
+      source_row,
+      expanded_source_row_to_expanded_dest_row,
+      total_rows_before_expert,
+      attr_mask,
+      num_rows,
+      k,
+      batch_size);
+}
+
+#define SOFTMAX_KERNEL(ITEMS_PER_THREAD)                                \
+  block.x /= ITEMS_PER_THREAD;                                          \
+  assert(block.x <= 1024);                                              \
+  if (is_half2) {                                                       \
+    if (grid.x % 4 == 0) {                                              \
+      grid.x /= 4;                                                      \
+      softmax_kernel_v5_half2<__half, ITEMS_PER_THREAD, 4>              \
+          <<<grid, block, 0, stream>>>(reinterpret_cast<half*>(buffer), \
+                                       (const half*)attr_mask,          \
+                                       batch_size,                      \
+                                       head_num,                        \
+                                       seq_len_1,                       \
+                                       seq_len_2,                       \
+                                       (const half)scalar);             \
+    } else {                                                            \
+      softmax_kernel_v4_half2<__half, ITEMS_PER_THREAD>                 \
+          <<<grid, block, 0, stream>>>(reinterpret_cast<half*>(buffer), \
+                                       (const half*)attr_mask,          \
+                                       batch_size,                      \
+                                       head_num,                        \
+                                       seq_len_1,                       \
+                                       seq_len_2,                       \
+                                       (const half)scalar);             \
+    }                                                                   \
+  } else {                                                              \
+    softmax_kernel_v4<ITEMS_PER_THREAD, T>                              \
+        <<<grid, block, 0, stream>>>(buffer,                            \
+                                     buffer_src,                        \
+                                     attr_mask,                         \
+                                     batch_size,                        \
+                                     head_num,                          \
+                                     seq_len_1,                         \
+                                     seq_len_2,                         \
+                                     scalar);                           \
+  }
+
+template <typename T>
+void invokeMaskedSoftMax(T* buffer,
```

Comment:
invokeMaskedSoftMax -> InvokeMaskedSoftMax. The following functions have the same problem

Reply:
done